aggregator/src/test/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayActionTest.java

```diff
@@ -0,0 +1,203 @@
+        final ReplayAction a = b.getAction(ReplayAction.class);
+        return ACL.impersonate(User.get(user).impersonate(), new NotReallyRoleSensitiveCallable<Boolean,RuntimeException>() {
+            @Override public Boolean call() throws RuntimeException {
+                return a.isEnabled();
+            }
+        });
+    }
+
+    @Test public void loadStep() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        FilePath f = r.jenkins.getWorkspaceFor(p).child("f.groovy");
+        f.write("echo 'original loaded text'", null);
+        p.setDefinition(new CpsFlowDefinition("node {load 'f.groovy'}", true));
+        WorkflowRun b1 = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("original loaded text", b1);
+        WorkflowRun b2 = (WorkflowRun) b1.getAction(ReplayAction.class).run("echo 'trying edits'\nnode {load 'f.groovy'}", Collections.singletonMap("Script1", "echo 'new loaded text'")).get();
+        r.assertBuildStatusSuccess(b2);
+        r.assertLogContains("trying edits", b2);
+        r.assertLogContains("new loaded text", b2);
+        System.out.println(new XmlFile(new File(b2.getRootDir(), "build.xml")).asString());
+        String diff = b2.getAction(ReplayAction.class).getDiff();
+        assertThat(diff, containsString("+echo 'trying edits'"));
+        assertThat(diff, containsString("Script1"));
+        assertThat(diff, containsString("-echo 'original loaded text'"));
+        assertThat(diff, containsString("+echo 'new loaded text'"));
+        System.out.println(diff);
+        // TODO test multiple loads
+        // TODO load after restart
+        // TODO test replay of replay
+    }
+
+    @Test public void cli() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        FilePath f = r.jenkins.getWorkspaceFor(p).child("f.groovy");
+        f.write("'original text'", null);
+        p.setDefinition(new CpsFlowDefinition("node {def t = load 'f.groovy'; echo \"got ${t}\"}", true));
+        WorkflowRun b1 = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("got original text", b1);
+        assertEquals(0, new CLICommandInvoker(r, "replay-pipeline").withStdin(IOUtils.toInputStream("node {def t = load 'f.groovy'; echo \"received ${t}\"}")).invokeWithArgs("p").returnCode());
+        r.waitUntilNoActivity();
+        WorkflowRun b2 = p.getLastBuild();
+        assertEquals(2, b2.getNumber());
+        r.assertLogContains("received original text", b2);
+        assertEquals(0, new CLICommandInvoker(r, "replay-pipeline").withStdin(IOUtils.toInputStream("'new text'")).invokeWithArgs("p", "-n", "1", "-s", "Script1").returnCode());
+        r.waitUntilNoActivity();
+        WorkflowRun b3 = p.getLastBuild();
+        assertEquals(3, b3.getNumber());
+        // Main script picked up from #1, not #2.
+        r.assertLogContains("got new text", b3);
+    }
+
+}
```

Comment:
<3 those comments, it's enough to make me happy and I'm on the extreme end about wanting comments.